src/os/bluestore/KernelDevice.cc

```diff
@@ -362,11 +362,17 @@ void KernelDevice::_aio_thread()
 	io_since_flush.store(true);
 
 	int r = aio[i]->get_return_value();
+	if (r < 0) {
+	  derr << __func__ << " finished aio " << aio[i] << " r " << r
+	       << " ioc " << ioc
+	       << " with " << (ioc->num_running.load() - 1)
+	       << " aios left" << dendl;
+	  assert(r >= 0);
+	}
 	dout(10) << __func__ << " finished aio " << aio[i] << " r " << r
```

Comment:
@joscollin at first, for dout(10) when the error happens this log with 10 level not always prints, because opening low level log impacts the performance, for the second, in order to avoid printing log twice when we exactly opened log of this level, i just handle the error before dout(10), how do you think of it ?

Reply:
just change the dout(10) to dout(r < 0 ? -1 : 10).  (dout(-1) is equivalent to derr)